### PR TITLE
Work on the todo changes in the code

### PR DESCRIPTION
**Pull request by KyrylSydorov**

### CODE REVIEW

File: Source/Portal/PortalGunAndComponents/UamsuPortalGun.cpp [hunk 1]
```diff
@@ -24,21 +24,24 @@ UamsuPortalGun::UamsuPortalGun()
...
+PRAGMA_DISABLE_OPTIMIZATION
 FHitResult UamsuPortalGun::GetAimedHitResult(float InCheckDistance, ECollisionChannel InCollisionChannel) const
 {
 	FVector ViewLocation = FVector::ZeroVector;
 	FRotator ViewRotation = FRotator::ZeroRotator;
 
-	APlayerController* PlayerController = GetOwner<APlayerController>();
+	// TODO It does not get a playerControler, find the way u get ht eplayer constoler
+	AActor* ActorOwner = GetOwner<AActor>();
+	APawn* PawnOwner = Cast<APawn>(ActorOwner);
```
Comment (KyrylSydorov) [inline Source/Portal/PortalGunAndComponents/UamsuPortalGun.cpp:35]:
```suggestion
	APawn* PawnOwner = GetOwner<APawn>();
	if (!ensure(IsValid(PawnOwner)))
	{
		return { };
	}
```

File: Source/Portal/Portals/amsuPortal.cpp [hunk 2]
```diff
@@ -35,14 +35,24 @@ void AamsuPortal::BeginPlay()
...
 void AamsuPortal::OnEndOverlap(UPrimitiveComponent* OverlappedComponent, AActor* OtherActor,
 	UPrimitiveComponent* OtherComp, int32 OtherBodyIndex)
 {
 }
 
+void AamsuPortal::Teleport(AActor* InteractedActor)
+{
+	FVector OutTeleportDistance(0.f, 200.f, 0.f);
+	FVector TeleportLocation = AnotherPortal->GetActorLocation();
+	InteractedActor->SetActorLocation(OutTeleportDistance + TeleportLocation);
```
Comment (KyrylSydorov) [inline Source/Portal/Portals/amsuPortal.cpp:53]:
```suggestion
	InteractedActor->SetActorLocation(OutTeleportDistance + TeleportLocation);
	// Direction
```

File: Source/Portal/Portals/amsuPortal.h [hunk 3]
```diff
@@ -34,4 +34,10 @@ class PORTAL_API AamsuPortal : public AActor
 	UFUNCTION()
 	void OnEndOverlap(UPrimitiveComponent* OverlappedComponent, AActor* OtherActor, UPrimitiveComponent* OtherComp,
 		int32 OtherBodyIndex);
+
+	TObjectPtr<AamsuPortal> AnotherPortal;
```
Comment (KyrylSydorov) [inline Source/Portal/Portals/amsuPortal.h:38]:
```suggestion
	UPROPERTY()
	TObjectPtr<AamsuPortal> AnotherPortal; 
```

**Comment by AmaterasuFury** [comment]
fixed

**Pull request closed without merging**